jkube-kit/config/image/src/test/java/org/eclipse/jkube/kit/config/image/build/HealthCheckConfigTest.java

```diff
@@ -13,146 +13,95 @@
  */
 package org.eclipse.jkube.kit.config.image.build;
 
-import org.junit.Assert;
-import org.junit.Test;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
 /**
  * Tests the health check configuration
  */
-public class HealthCheckConfigTest {
-
-    @Test
-    public void testGoodHealthCheck1() {
-        HealthCheckConfiguration.builder()
-                .cmd(Arguments.builder().shell("exit 0").build())
-                .build()
-                .validate();
-    }
-
-    @Test
-    public void testGoodHealthCheck2() {
-        HealthCheckConfiguration.builder()
-                .cmd(Arguments.builder().shell("exit 0").build())
-                .retries(1)
-                .build()
-                .validate();
-    }
-
-    @Test
-    public void testGoodHealthCheck3() {
-        HealthCheckConfiguration.builder()
-                .cmd(Arguments.builder().shell("exit 0").build())
-                .retries(1)
-                .interval("2s")
-                .build()
-                .validate();
-    }
 
-    @Test
-    public void testGoodHealthCheck4() {
-        HealthCheckConfiguration.builder()
-                .cmd(Arguments.builder().shell("exit 0").build())
-                .retries(1)
-                .interval("2s")
-                .timeout("3s")
-                .build()
-                .validate();
-    }
-
-    @Test
-    public void testGoodHealthCheck5() {
-        HealthCheckConfiguration.builder()
-                .mode(HealthCheckMode.cmd)
-                .cmd(Arguments.builder().shell("exit 0").build())
-                .retries(1)
-                .interval("2s")
-                .timeout("3s")
-                .startPeriod("30s")
-                .build()
-                .validate();
-    }
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```

Comment:
Why is this required? I'm not sure if tests are dependent on each other.

Reply:
These tests are not dependent on each other. If we don't add this order, `Good Health Check 7` and `Bad Health Check 7` are executed first and after that rest of the tests are executed which doesn't look good. So, executing these in this order, all `Good health check tests` are executed first and after that, `Bad health check tests` are executed.

Reply:
I'm not sure if this is a good reason for making tests ordered.

Reply:
I'll remove it

Reply:
The ideal option would have been to rename the tests, `Good Health Check 7` is a very good example of a bad name for a test :sweat_smile: 
I have no idea what these tests are testing, nor what the spec says about the code they are verifying.